client/src/pages/review/UpdateReviewForm.tsx

```diff
@@ -1,76 +1,103 @@
-              <TextInputField errors={errors} field="title" register={register} label="Tytuł" defaultValue={review.title+""} />
-              <TextInputBox rows={10} errors={errors} field="description" register={register} label="Komentarz" defaultValue={review.description+""} />
-              <input type="hidden" {...register("idReview")} value={review.id}/>
-              <input type="hidden" {...register("idBook")} value={book.id}/>
-              <input type="hidden" {...register("rating")} value={value+""}/>
-              <Button variant="contained" type="submit"  sx={{ margin: 1}}>
-                Uaktualnij
-              </Button>
-              <Button variant="contained" onClick={() => handleClose()} sx={{ margin: 1 }}>
-                Anuluj
-              </Button>
-            </Paper>
-          </Box>
-        </form>
-      </Box>
-    );
+function UpdateReviewForm({
+  review,
+  book,
+  handleClose,
+  refetch,
+}: {
+  review: ReviewViewModelType;
+  book: BookViewModelType;
+  handleClose: () => void;
+  refetch: () => Promise<unknown>;
+}) {
+  const [value, setValue] = React.useState<number | null>(review.rating);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<UpdateReviewType>({
+    resolver: zodResolver(UpdateReview),
+  });
+  const mutation = useMutation({
+    mutationFn: updateReview,
+    onSuccess: async () => {
+      handleClose();
+      await refetch();
+      window.location.reload();
+    },
+    onError: (e: Error) => {
+      console.error(e);
+    },
+  });
+  const onSubmit: SubmitHandler<UpdateReviewType> = async (data) => {
+    data.rating = value == null ? 0 : value;
+    data.idReview = review.id;
+    data.idBook = book.id;
+    mutation.mutate(data);
+  };
+
+  return (
+    <Box sx={{ mt: 0 }}>
+      <form onSubmit={handleSubmit(onSubmit)} style={{ display: 'flex', justifyContent: 'center' }}>
+        <Box
+          sx={{
+            width: 'auto',
+            textAlign: 'left',
+          }}>
+          <Paper sx={{ p: 2, width: '100%' }} elevation={3}>
+            <Rating
+              name="rating"
+              value={value}
+              onChange={(_, newValue) => {
+                setValue(newValue);
+              }}
+            />
+            <TextInputField
+              errors={errors}
+              field="title"
+              register={register}
+              label="Tytuł"
+              defaultValue={review.title + ''}
+            />
+            <TextInputBox
+              rows={10}
+              errors={errors}
+              field="description"
+              register={register}
+              label="Komentarz"
+              defaultValue={review.description + ''}
+            />
+            <input type="hidden" {...register('idReview')} value={review.id} />
+            <input type="hidden" {...register('idBook')} value={book.id} />
+            <input type="hidden" {...register('rating')} value={value + ''} />
```

Comment:
to lepiej zastąpić domyślnymi wartościami na poziomie `useForm`
```js
useForm({
  defaultValues: {
    firstName: '',
    lastName: ''
  }
})
```
https://react-hook-form.com/docs/useform#defaultValues